### PR TITLE
Updated progressbar component to fix a warning of a deprecated method

### PR DESCRIPTION
**Pull request by markusamuel**
image

### CODE REVIEW

**Comment by mark-yacoub** [comment]
interesting but wrong :p in the right direction but not how it's to be solved.
I'm not even sure how did this even work.
node_modules should be in .gitignore. if it's not, then it must be added there. node_modules get overwritten all the time so this fix will disappear at the first clean.
what we do here instead is to use `patch-package`.
This allows you to patch any package on yarn install, guaranteeing a fix at all times.
use `patch-package` (add it as a dev dep). the docs are quite good but lmk if you need help.

File: patches/react-native-progress-bar-animated+1.0.6.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+index 29a0f82..b5e6774 100644
+--- a/node_modules/react-native-progress-bar-animated/src/AnimatedProgressBar.js
++++ b/node_modules/react-native-progress-bar-animated/src/AnimatedProgressBar.js
+@@ -26,8 +26,8 @@ class ProgressBar extends React.Component {
+     }
+   }
+ 
+-  componentWillReceiveProps(props) {
+-    if (props.value !== this.state.progress) {
++  static getDerivedStateFromProps (props,state) {
```
Comment (mark-yacoub) [inline patches/react-native-progress-bar-animated+1.0.6.patch:13]: better, but 2 things. one of which is optional.
in 5 years, we will not remember why we are patching this, so add a comment on top of the new code explaining what error/warning are you fixing. so when the packages get updated and we wanna remove the patch, we know if it's safe to delete the patch. and be specific on how to repro the warning you're fixing.
so like `removing btingan warning that appears when the progress bar is rendered`

2. look into the package github issues and see if someone has mentioned. if so, you'll either find people suggesting fixes, or someone sent a fix PR. for either cases, add a comment here with the link that has the fix, basically referencing where did you get this fix from.

Comment (markusamuel) [reply patches/react-native-progress-bar-animated+1.0.6.patch:13]: comment added

File: patches/react-native-progress-bar-animated+1.0.6.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
+This patch is to remove the Warning "componentWillReceiveProps has been renamed and not recommended for use" and "useNativeDriver was not specified"
```
Comment (mark-yacoub) [inline patches/react-native-progress-bar-animated+1.0.6.patch:1]: Not sure if this compiles but if it does eshta 3alek

**Review by mark-yacoub — APPROVED** [review]

**Pull request merged**